Build error on release builds

On mingw, gcc and clang build fails in release mode with the following error:
> src/codegen.cpp:533: error: third operand to the conditional operator is of type 'void', but the second operand is neither a throw-expression nor of type 'void'

However despite complaining at line 533, the error is actually at line 534 with the use of `Unreachable()` In debug mode this definition is used https://github.com/thelink2012/gta3sc/blob/master/src/cpp/contracts.hpp#L45 which builds without issue. But in release mode this definition is used https://github.com/thelink2012/gta3sc/blob/master/src/cpp/contracts.hpp#L52 causing this error.

--- a/src/cpp/contracts.hpp
+++ b/src/cpp/contracts.hpp
@@ -49,6 +49,7 @@ struct unreachable_exception : std::runtime_error
 
 #define Expects(cond)     void(0)
 #define Ensures(cond)     void(0)
-#define Unreachable()     (std::terminate(), throw 0)
+#define Unreachable()     \
+    (throw std::runtime_error("Unreachable code reached"))
 
 #endif

